README.md

```diff
@@ -0,0 +1,390 @@
+# `pam_pkcs11`
+
+This project is a Puppet module to manage *pam_pkcs11*, a PAM module developed
+as part of the [OpenSC project] for advanced authentication with smart cards
+and other PKCS #11 cryptographic modules.
+
+[OpenSC project]: https://github.com/OpenSC/OpenSC/wiki
+
+## Table of Contents
+
+1. [Setup - The basics of getting started with pam_pkcs11](#setup)
+  * [What pam_pkcs11 affects](#what-pam_pkcs11-affects)
+  * [Setup requirements](#setup-requirements)
+  * [Beginning with pam_pkcs11](#beginning-with-pam_pkcs11)
+1. [Usage - Configuration options and additional functionality](#usage)
+1. [Reference - An under-the-hood peek at what the module is doing and how](#reference)
+1. [Limitations - OS compatibility, etc.](#limitations)
+1. [Development - Guide for contributing to the module](#development)
```

Comment:
I really like this structure! :100: 

Maybe the last point could be replaced by a link to the `CONTRIBUTING.md` file?

Reply:
Yes done that.